pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java

```diff
@@ -896,9 +896,36 @@ public void testSearch()
     assertNoError(jsonNode);
   }
 
+  @Test(enabled = false)
+  public void testBetween() throws Exception {
+    String sqlQuery = "SELECT COUNT(*) FROM mytable WHERE ArrDelay BETWEEN 10 AND 50";
+    JsonNode jsonNode = postQuery(sqlQuery);
+    assertNoError(jsonNode);
+    assertEquals(jsonNode.get("resultTable").get("rows").get(0).get(0).asInt(), 18572);
+
+    String explainQuery = "EXPLAIN PLAN FOR " + sqlQuery;
+    jsonNode = postQuery(explainQuery);
+    assertNoError(jsonNode);
+    String plan = jsonNode.get("resultTable").get("rows").get(0).get(1).asText();
+    // Ensure that the BETWEEN filter predicate is converted to >= and <=
+    Assert.assertFalse(plan.contains("BETWEEN"));
+    Assert.assertTrue(plan.contains(">="));
+    Assert.assertTrue(plan.contains("<="));
+
+    // No rows should be returned since lower bound is greater than upper bound
+    sqlQuery = "SELECT COUNT(*) FROM mytable WHERE ARRAY_TO_MV(RandomAirports) BETWEEN 'SUN' AND 'GTR'";
+    jsonNode = postQuery(sqlQuery);
```

Comment:
The test is currently disabled because this query fails with an assertion error during query compilation (in the `SqlToRelConverter` phase) in the test runner where assertions are enabled (`-ea`). The same query passes on the quickstart though since assertions are disabled by default at runtime. The reason for this is that Calcite expects `BETWEEN` to always be rewritten before being converted to `RexNode` and this assumption is broken in the `BetweenConvertlet` added to `PinotConvertletTable` here which will create a `RexNode` with the `SqlBetweenOperator` when the first operand is a call to `ARRAY_TO_MV`. This logic is implemented in Calcite [here](https://github.com/apache/calcite/blob/60e0a3f441a009e55a36cac192253a436bec3f6d/core/src/main/java/org/apache/calcite/sql/fun/SqlBetweenOperator.java#L117) and [here](https://github.com/apache/calcite/blob/60e0a3f441a009e55a36cac192253a436bec3f6d/core/src/main/java/org/apache/calcite/rex/RexCall.java#L93). This isn't actually a problem for Pinot though since we do have a `BETWEEN` filter predicate implementation.

Reply:
Fixed in https://github.com/apache/pinot/pull/14135/commits/f0d97d35b086fdb95dd8e0be346cf56829cc99cb.